### ISSUE TITLE
about:blank instead of tabs

### ISSUE THREAD

**Issue by Mornon** [hrj/sloth#6]
I'm trying Sloth on Chrome 66.0.3359.117, but when I opened it all my tabs went about:blank - Chrome didn't load them, but the original adddress was lost.

**Comment by hrj** [comment]
That is unlikely to be caused by sloth though. The code in sloth is simple.. it just marks each tab as inactive. It could be another extension or some bug in Chrome itself.

I will leave this issue open for now, to collect more feedback.

**Comment by Mornon** [comment]
I tried disabling all extension beside Sloth and it still happened; not sure if it's a Chrome bug or what else.

**Comment by hrj** [comment]
@Mornon  Nobody else has confirmed this bug yet. Is it still happening for you? And if yes, then can you check whether disabling Sloth stops the problem?

**Comment by Mornon** [comment]
I just tried with a brand new profile - no settings, no extensions, just a new profile, and it still does happen, also if just for pinned tabs: I pinned a tab, when I reopened Chrome it was blank; disabling Sloth let the tab load normally.

**Comment by hrj** [comment]
Gotcha; re-opening the issue.

Please describe your chrome version and OS version, since this is not happening for other users.

**Comment by Mornon** [comment]
Windows 10 Pro 1803 17134.191.

**Comment by Mornon** [comment]
I did a bit more testing: it does happen on both pinned and not pinned tabs, it just loads the current tab (so in my previous test it worked on the unpinnd tab because it was the only one). The tab that starts selected gets loaded, the others are just about:blank.

Strange thing, it did not happen with https://chrome.google.com/webstore/search/Sloth?_category=extensions - this page gets loaded also if not selected, so it didn't get turned to blank, but it also didn't get lazy loaded, it just loads normally. No idea why.

**Comment by q2p** [comment]
It's a few years too late, but the problem is essentially that if you unload a tab in Chrome before it performs some sort of internal loading within Chrome it will get permanently replaced with about:blank, wiping out all that tab's history.
The bug is less prevalent nowadays (I guess due to my PC having more RAM), but it still happens to some tabs if you crash Chrome with ~10k tabs or after a Windows update.
I found no solution to this myself, so I just backup tabs to a server and look for about:blank tabs periodically with an extension.
I tried doing "don't unload tabs unless their state is loaded" but then it just loads everything. I tried doing "don't unload tabs unless N amount of seconds pass" but it only partially solves the problem, some tabs still get stuck in a limbo state and get turned into about:blanks after subsequent restart.
And from my experience reporting bugs to Chrome it usually takes a decade to resolve, so I doubt there will be a fix from the Chrome team.

**Comment by hrj** [comment]
@q2p Thanks for your analysis. FWIW, I have never faced the issue myself. 

For users who have faced the issue even once, I urge them to report upstream, even if it takes them a while to address. If you do, please paste the issue link here so that we can all up-vote it.

**Comment by Luckz** [comment]
@q2p 
> if you crash Chrome with ~10k tabs or after a Windows update.
I found no solution to this myself, so I just backup tabs to a server and look for about:blank tabs periodically with an extension.

For more than at most a few hundred tabs, Chrome's kind of unsuitable anyway, eh? Unless they're all in a single window, it will always load one or even a few per window at startup, start loading them before the ad-blocking (or VPN/proxy extension) is enabled, and on quit / crash it happens *too often* that you can't fully restore all dozens of windows, etc.

Old old old Firefox (XUL) had the powerful Session Manager by Michael Kraft that could  save/restore individual windows, and actually save full tabs with their history etc, but besides Vivaldi no Chromium has attempted things like that, and when I tried it Vivaldi had such a messy codebase (?) that performance broke at the seams after a low 3 digit number of tabs (while other Chrome forks can get into 4 digits of unloaded tabs but have huge issues  with startup/shutdown).

**tldr** : I think among contemporary browsers only heavily tweaked Firefox could handle your 10k tabs


When you say backup *tabs* to a server, you essentially just mean URLs (~= _bookmarks_), right? Since _tabs_ have things like navigation history (forwards, backward), and scroll state, etc. Saving those would be back to the above mentioned XUL Firefox.
